go-controller/pkg/node/gateway_shared_intf_linux.go

```diff
@@ -1,17 +1,78 @@
+// +build linux
+
 package node
 
 import (
 	"fmt"
 	"net"
+	"strings"
 
 	"github.com/coreos/go-iptables/iptables"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
-
 	kapi "k8s.io/api/core/v1"
 	"k8s.io/klog"
 	utilnet "k8s.io/utils/net"
 )
 
+// Creates br-local OVS bridge on the node and adds ovn-k8s-gw0 port to it with the
+// appropriate IP and MAC address on it. All the traffic from this node's hostNetwork
+// Pod towards cluster service ip whose backend is the node itself is forwarded to the
+// ovn-k8s-gw0 port after SNATing by the OVN's distributed gateway port.
+func setupLocalNodeAccessBridge(nodeName string, subnet *net.IPNet) error {
+	localBridgeName := "br-local"
+	_, stderr, err := util.RunOVSVsctl("--may-exist", "add-br", localBridgeName)
+	if err != nil {
+		return fmt.Errorf("failed to create bridge %s, stderr:%s (%v)",
+			localBridgeName, stderr, err)
+	}
+
+	_, _, err = bridgedGatewayNodeSetup(nodeName, localBridgeName, localBridgeName,
+		util.LocalNetworkName, true)
+	if err != nil {
+		return fmt.Errorf("failed while setting up local node access bridge : %v", err)
+	}
+
+	_, err = util.LinkSetUp(localBridgeName)
+	if err != nil {
+		return err
+	}
+
+	macAddress := util.IPAddrToHWAddr(net.ParseIP(util.V4DnatAndSnatNextHop)).String()
```

Comment:
There is a difference in how these two subnets are being used. The semantics are different as well.

1. The subnet used for local gateway mode is per-node. Strictly speaking it only needs two IPs. So, it could be just 169.254.33.0/30. All we need is 169.254.33.1 and 169.254.33.2.

2. The subnet size for the logical switch that the distributed gateway port connects to is large. I have used /20, and with that prefix length we can support upt to 2^12 (4000+) nodes K8s cluster.

So, we can't merge those two subnets.

Reply:
Do either of these work?

1. set the values based on which mode is used OR
2. not set these values in util package and keep them local to the node package you are using (like gw local does)

Reply:
@trozet the reason it is in util package is because I need those values from 
1. 'pkg/ovn' - the files under this package sets up the logical constructs using these values in ovnkube-master daemon
2. 'pkg/node' - the files under this package sets up the physical constructs (br-local, ovn-k8s-gw0, and so on) using these values in ovnkube-node daemon